src/Cases.py

```
@@ -375,6 +375,14 @@ def read_input(self, xml):
       self.raiseAnError('No <dispatch> node was provided in the <Case> node!')
     if self._time_discretization is None:
       self.raiseAnError('<time_discretization> node was not provided in the <Case> node!')
+    cores_requested = self.innerParallel * self.outerParallel
+    if cores_requested > 1:
+      # check to see if the number of processes available can meet the request
+      detected = os.cpu_count() - 1 # -1 to prevent machine OS locking up
+      if detected < cores_requested:
+        self.raiseAWarning('System may be overloaded and greatly increase run time! ' +
```

Just a comment for consideration:

On my Windows machine, requesting more cores than you have available results in an error and RAVEN will not run, not sure what the behavior is on Mac or Linux. We could raise an error here or raise a warning and wait for RAVEN to throw an error later, either way is probably fine. I think the important thing is to inform the user that they requested more cores than the machine has available. If that particular error is thrown later, the user will have an idea as to why the simulations were not run.

---

Yeah, I don't love how this is handled generally speaking; I think handling it better in RAVEN may lead to sensible error messages so we don't duplicate that effort in HERON.